src/entities/jsonld.ts

```diff
@@ -1,25 +1,66 @@
 import { EDC_CONTEXT } from "./context";
+import jsonld from "jsonld";
+
+export async function expand<T extends JsonLdObject>(body: any, newInstance: (() => T)): Promise<T> {
+  const expanded = await jsonld.expand(body);
+  var instance = Object.assign(newInstance(), expanded[0]);
+  instance._compacted = body;
+  return instance;
+};
+
+export async function expandArray<T extends JsonLdObject>(body: any, newInstance: (() => T)): Promise<T[]> {
+  const expanded = await jsonld.expand(body);
+  return (expanded as Array<any>).map((element, index) => {
+    var instance = Object.assign(newInstance(), element);
+    instance._compacted = body[index];
+    return instance;
+  });
+}
 
 export class JsonLdObject {
 
   [propertyName: string]: any;
+  /*
+   * this _compacted field is a workaround for a bug existing in the json-ld
+   * which causes the EDC offer ids to be "truncated". For further info, take a
+   * look at https://github.com/digitalbazaar/jsonld.js/issues/523
+   * Once that issue will be fixed, we can get rid of the _compacted and all
+   * its usages.
+   */
+  set _compacted(_compacted: any) {
+    this.__compacted = _compacted;
+  }
+
+  get _compacted() {
+    return this.__compacted;
+  }
```

Comment:
@ndr-brt @OlfaBensoussia could you shortly explain in this PR how the workaround works? I see we are favouring it over other namespace values, but I feel I'm missing something 😅